transport/src/main/java/io/netty5/channel/socket/nio/ByteBufferCollector.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty5.channel.socket.nio;
+
+import io.netty5.buffer.api.Buffer;
+import io.netty5.channel.ChannelOutboundBuffer;
+import io.netty5.util.concurrent.FastThreadLocal;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+final class ByteBufferCollector implements ChannelOutboundBuffer.MessageProcessor<RuntimeException> {
```

Comment:
How about some tests for this class? Should in theory be easier now that it's a separate thing.

Reply:
yep let me port over the "old" ones